Source/EventAnalysis/KTOverlappingTrackClustering.cc

```diff
@@ -0,0 +1,406 @@
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    /*
+    bool KTOverlappingTrackClustering::DoTheyCross(KTProcessedTrackData& Track1, KTProcessedTrackData& Track2)
+    {
+        // if the start time or the end time of track 2 is between start and end time of track 1
+        bool condition1 = Track2.GetStartTimeInRunC() < Track1.GetEndTimeInRunC() and Track2.GetStartTimeInRunC() > Track1.GetStartTimeInRunC();
+        bool condition2 = Track2.GetEndTimeInRunC() < Track1.GetEndTimeInRunC() and Track2.GetEndTimeInRunC() > Track1.GetStartTimeInRunC();
+
+        // and they cross in frequency this way
+        bool cross_condition1 = Track2.GetStartFrequency() < (Track1.GetStartFrequency() + Track1.GetSlope() * (Track2.GetStartTimeInRunC() - Track1.GetStartTimeInRunC()));
+        bool cross_condition2 = Track2.GetEndFrequency() > (Track1.GetStartFrequency() + Track1.GetSlope() * (Track2.GetEndTimeInRunC() - Track1.GetStartTimeInRunC()));
+
+        // or the other way around
+        bool cross_condition3 = Track2.GetStartFrequency() > (Track1.GetStartFrequency() + Track1.GetSlope() * (Track2.GetStartTimeInRunC() - Track1.GetStartTimeInRunC()));
+        bool cross_condition4 = Track2.GetEndFrequency() < (Track1.GetStartFrequency() + Track1.GetSlope() * (Track2.GetEndTimeInRunC() - Track1.GetStartTimeInRunC()));
+
+
+        if ((condition1 or condition2) and ((cross_condition1 and cross_condition2) or (cross_condition3 and cross_condition4)))
+        {
+            return true;
+        }
+
+        // the same as above but for the inverted track combination
+        condition1 = Track1.GetStartTimeInRunC() < Track2.GetEndTimeInRunC() and Track1.GetStartTimeInRunC() > Track2.GetStartTimeInRunC();
+        condition2 = Track1.GetEndTimeInRunC() < Track2.GetEndTimeInRunC() and Track1.GetEndTimeInRunC() > Track2.GetStartTimeInRunC();
+   
+        // and they cross in frequency this way
+        cross_condition1 = Track1.GetStartFrequency() < (Track2.GetStartFrequency() + Track2.GetSlope() * (Track1.GetStartTimeInRunC() - Track2.GetStartTimeInRunC()));
+        cross_condition2 = Track1.GetEndFrequency() > (Track2.GetStartFrequency() + Track2.GetSlope() * (Track1.GetEndTimeInRunC() - Track2.GetStartTimeInRunC()));
+        
+        // or the other way around
+        cross_condition3 = Track1.GetStartFrequency() > (Track2.GetStartFrequency() + Track2.GetSlope() * (Track1.GetStartTimeInRunC() - Track2.GetStartTimeInRunC()));
+        cross_condition4 = Track1.GetEndFrequency() < (Track2.GetStartFrequency() + Track2.GetSlope() * (Track1.GetEndTimeInRunC() - Track2.GetStartTimeInRunC()));
+   
+   
+        if ((condition1 or condition2) and ((cross_condition1 and cross_condition2) or (cross_condition3 and cross_condition4)))
+        {   
+            return true;
+        }
+
+
+        return false;
+    }*/
+
+    void KTOverlappingTrackClustering::EmitTrackCandidates()
```

Comment:
Variable names to fix: `LineIsTrack`

`LineIsTrack` should be initialized when declared